pallets/deeper-node/src/lib.rs

```diff
@@ -58,16 +68,24 @@ decl_module! {
         fn deposit_event() = default;
 
         #[weight = 10_000]
-        pub fn register_device(origin, ip: Vec<u8>, country: u16) -> DispatchResult {
+        pub fn register_device(origin, ip: Vec<u8>, country: Vec<u8>) -> DispatchResult {
```

Comment:
I would suggest modify register_device a little bit so that it can also able to modify country and IP if it's not the first time to register, and only reserve MIN_LOCK_AMT the first time.